### DISCUSSION TITLE
Reliable build native binary for raspberry pi?

### DISCUSSION THREAD

**Discussion by DigitalMediageek** [oracle/graal#9003] [Q&A]
Hi,

is there a way to reliably build native binaries running on raspberry pi? The best raspi I can offer is 4B with 4GB of ram. Building native binaries on it is either hell slow or breaks with out of memory error. The only working way I found is using QEMU, which is even slower then on raspi. I also tried using ARM64 VMs in the cloud, but they are either unavailable or to expensive. The best guess for having enough ram and lower building times would be cross compilation from AMD64. But from following existing discussions on this subject I understand that this is a complex, currently not working path. 

Any hints on that? Has someone managed to build native binaries for raspi with reasonable effort?

Best Regards

**Comment by alina-yur** [comment]
Hi @DigitalMediageek. 4 GB should be enough for small and medium-size apps, even Petclinic would compile with just 2 GB (granted, slower in that case). Seems like you might be compiling a bigger project. The options that I see either getting a bigger machine, or looking into building elsewhere, or trying the  -`O0` / `-Ob` build flags if you can trade off peak throughput.
